packages/core/core/src/index.ts

```diff
@@ -20,3 +20,25 @@ export const createStrapi = (options: Partial<StrapiOptions> = {}): Core.Strapi
 
   return strapi;
 };
+
+// Augment Koa query type based on Strapi query middleware
+
+import * as qs from 'qs';
+
+declare module 'koa' {
```

Comment:
can't we reuse the IParseOptions from qs ?

Reply:
Where would we use that in this file? :thinking:

Reply:
If this is related to the middlewares/query.ts change

`qs.parse` has 2 definitions, one that returns `{ [key: string]: unknown }` and the other returns `ParsedQs`.
The return type will be inferred based on the params passed to the function.
- If `IParseOptions & { decoder?: never | undefined }` is passed, then it'll resolve to `{ [key: string]: unknown }`
- If only `IParseOptions` is passed, then it'll resolve to `ParsedQs`

So here we had the choice of setting Config to `IParseOptions & { decoder?: never | undefined } | IParseOptions` or `Parameters<typeof qs.parse>[1]`. The first one is more atomic/readable, the other one is more maintainable as it'll automatically adapt to future signatures.

I'm OK with both :shrug:

Reply:
It'll default to that through inference in some case such as here https://github.com/strapi/strapi/pull/20659/files#diff-3fcd641ac6072f7f3c6cde350394972f7998e6bc022c38e0edf320e81ef78e38R30

It's kinda needed since `{ [key: string]: unknown ? }` is wider than `ParsedQs`. All in all it's just to be covered, but it's minor